### PR TITLE
Add validation of Dockerfile

### PR DESCRIPTION
**Pull request by Dufgui**
We validate Xml, in our continuous integration. That's why want to do the same on Dockerfile.
Until now we just validate the format, but the target idea is to add "Best practice" check in the validation.

### CODE REVIEW

File: docker-java-orchestration-core/src/main/java/com/alexecollins/docker/orchestration/DockerOrchestrator.java [hunk 1]
```diff
@@ -127,6 +127,19 @@ void build(final Id id) {
...
+        if (id == null) {
+            throw new IllegalArgumentException("id is null");
+        }
+        try {
+            FileValidator.validate(id, repo.src(id));
+        } catch (IOException e) {
+            throw new OrchestrationException(e);
+        }
+
+        snooze();
```
Comment (alexec) [inline docker-java-orchestration-core/src/main/java/com/alexecollins/docker/orchestration/DockerOrchestrator.java]: Can remove this line.

File: docker-java-orchestration-core/src/main/java/com/alexecollins/docker/orchestration/FileValidator.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Created with IntelliJ IDEA.
```
Comment (alexec) [inline docker-java-orchestration-core/src/main/java/com/alexecollins/docker/orchestration/FileValidator.java]: Could remove this default header.

File: docker-java-orchestration-core/src/main/java/com/alexecollins/docker/orchestration/FileValidator.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+ * User: guillaume_dufour
+ * Date: 07/01/15
+ * Time: 15:25
+ * To change this template use File | Settings | File Templates.
+ */
+public class FileValidator {
```
Comment (alexec) [inline docker-java-orchestration-core/src/main/java/com/alexecollins/docker/orchestration/FileValidator.java]: Maybe `DockerfileValidator`?

File: docker-java-orchestration-core/src/test/java/com/alexecollins/docker/orchestration/DockerOrchestratorIT.java [hunk 4]
```diff
@@ -99,4 +113,24 @@ public void testPush() throws Exception {
     public void testIsRunning() throws Exception {
         orchestrator.isRunning();
     }
+
+    @Test
+    public void testValidate() throws Exception {
+        boolean hasException = false;
```
Comment (alexec) [inline docker-java-orchestration-core/src/test/java/com/alexecollins/docker/orchestration/DockerOrchestratorIT.java]: Maybe two testing using: http://www.alexecollins.com/try-fail-catch-assert-junit-patttern/

File: docker-java-orchestration-core/src/test/java/com/alexecollins/docker/orchestration/DockerOrchestratorIT.java [hunk 5]
```diff
@@ -45,6 +49,16 @@ public void setUp() throws Exception {
                 "docker-java-orchestrator",
                 DockerOrchestrator.DEFAULT_FILTER,
                 new Properties());
+
+        wrongOrchestrator = new DockerOrchestrator(
```
Comment (alexec) [inline docker-java-orchestration-core/src/test/java/com/alexecollins/docker/orchestration/DockerOrchestratorIT.java]: Good stuff.

**Comment by alexec** [comment]
Great stuff. An excellent addition.

**Comment by Dufgui** [comment]
Thanks for merging my PR. I try to continue this work : 
- log all error before throw the error (already start in my repo)
- add warning when you don't respect best practice, etc... 

Are you ok with this goal ? can i finish it and push you a new PR?

**Comment by alexec** [comment]
Go for your life, but you'll want to merge master back into your fork.

**Pull request merged**